lib/process-services-cloud/src/lib/form/components/widgets/dropdown/dropdown-cloud.widget.ts

```diff
@@ -74,23 +80,111 @@ export class DropdownCloudWidgetComponent extends WidgetComponent implements OnI
+        const variableName = this.field?.variableConfig?.variableName;
+        const processVariables = this.field?.form?.processVariables;
+        const formVariables = this.field?.form?.variables;
+
+        const dropdownOptions = this.getOptionsFromVariable(processVariables, formVariables, variableName);
+
+        if (dropdownOptions) {
+            const formVariableOptions: FormFieldOption[] = this.getOptionsFromPath(dropdownOptions, optionsPath);
+            this.field.options = formVariableOptions;
+            this.resetInvalidValue();
+            this.field.updateForm();
+        } else {
+            this.handleError(`${variableName} not found`);
+            this.resetOptions();
+            this.variableOptionsFailed = true;
         }
-        this.updateOptions();
+    }
+
+    private getOptionsFromPath(data: any, path: string): FormFieldOption[] {
+        const optionsId = this.field?.variableConfig?.optionsId ?? this.defaultVariableOptionId;
+        const optionsLabel = this.field?.variableConfig?.optionsLabel ?? this.defaultVariableOptionLabel;
+
+        const properties = path.split('.');
+        const currentProperty = properties.shift();
+
+        if (!data.hasOwnProperty(currentProperty)) {
+            this.handleError(`${currentProperty} not found in ${JSON.stringify(data)}`);
+            this.variableOptionsFailed = true;
+            return [];
+        }
+
+        const nestedData = data[currentProperty];
+
+        if (Array.isArray(nestedData)) {
+            return this.getOptionsFromArray(nestedData, optionsId, optionsLabel);
+        }
+
+        return this.getOptionsFromPath(nestedData, properties.join('.'));
+    }
+
+    private getOptionsFromArray(nestedData: any[], id: string, label: string): FormFieldOption[] {
+        const options = nestedData.map(item => this.createOption(item, id, label));
+        const hasInvalidOption = options.some(option => !option);
+
+        if (hasInvalidOption) {
+            this.variableOptionsFailed = true;
+            return [];
+        }
+
+        this.variableOptionsFailed = false;
+        return options;
+    }
+
+    private createOption(item: any, id: string, label: string): FormFieldOption {
+        const option: FormFieldOption = {
+            id: item[id],
+            name: item[label]
+        };
+
+        if (!option.id || !option.name) {
+            this.handleError(`'id' or 'label' is not properly defined`);
+            return undefined;
+        }
+
+        return option;
+    }
+
+    private getOptionsFromVariable(processVariables: TaskVariableCloud[], formVariables: TaskVariableCloud[], variableName: string): TaskVariableCloud {
+        const processVariableDropdownOptions: TaskVariableCloud = this.getVariableValueByName(processVariables, variableName);
+        const formVariableDropdownOptions: TaskVariableCloud = this.getVariableValueByName(formVariables, variableName);
```

Comment:
`the ones from a form variable are ignored anyways` im not sure about that. we have two different properties of the form, `processVariables` and `variables` (formVariables), and they have their own values. Mapping give us ability to access mapped process variables (by input mapping) from the form definition, and use process variable values if they are.
and because we can have a case, where we don't use process variables (mapping), we need to check if we have some form variables available, that is why i have this line here (and also formVariables passed as parameter)